shapeshift/web
# fix: trade shows as completed after Tx broadcast

## Description

This 
- fixes the trades showing as completed right after Tx broadcast.
The reason for that is we were accessing the last ID in `txIdsByAssetId` whereas the latest Tx is actually the head.
- removes `selectLastTxStatusByAssetId` which was a hack to get the "last" (as in, the last ID in the txHistorySlice) in favor of selecting the status of the actual returned Tx from swapper with a new `selectTxStatusById`
- lowercases the tx address in in the `makeUniqueTxId` function, which makes a unique ID in the form of `accountId*txId*txAddress`:
https://github.com/shapeshift/web/blob/e463b2f394b7e59c3c21900f9d5a22b0cf660829/src/state/slices/txHistorySlice/utils.ts#L48-L50
This was needed because otherwise, we can't select a specific TxId in the slice, since we don't have the case-sensitive last part available anywhere.
- removes the now unused `selectTxIdsByAssetId` 
- adds a pending state icon in `<AssetToAsset />`

## Notice




## Pull Request Type


## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/1747

## Risk

Going from case-sensitive to lowercase in the last `txAddress` part of `accountId*txId*txAddress` shouldn't cause regressions, since it's just used for indexing and AFAIK we're not relying on the case-sensitive format anywhere. This is coming from the unchained WS payload on new message, and given we never needed to go from raw TxId to unique TxId in the aforementioned form anywhere, it's probably just something that we didn't have the need to lowercase before.

## Testing

1. Make a trade up to the "Confirm and Trade" screen - arrow should be shown
2. Click Confirm & Trade - a spinner should be shown instead
3. After MetaMask confirmation popup and when Tx is confirmed on the blockchain, a green check icon should be shown (see screenshot)
------------------------------------------------------------------------------>

## Screenshots (if applicable)

Didn't take a screenshot of the loading state with spinner and probably no point to spend another Tx on this for the sake of taking a screenshot, but this is what the step looks like after Tx confirmation:

<img width="401" alt="image" src="https://user-images.githubusercontent.com/17035424/168316855-00b14a24-dcb2-4e6f-a009-0e902a9dc7ca.png">
